### DISCUSSION TITLE
Example of vault-authentication-role used in vault-authentication project

### DISCUSSION THREAD

**Discussion by jeffery** [FriendsOfTerraform/modules#17] [Q&A]
Hi there,

I am looking for examples of what is in the `vault-authentication-role` role as per project reference https://github.com/FriendsOfTerraform/vault-authentication

Any tips or documentation would be helpful.

Thanks,
Jeff

**Comment by petersin0422** [comment]
Hello Jeff,

I am assuming you are referring to this `sts_role_arn  = "arn:aws:iam::111122223333:role/vault-authentication-role"`? This refers to an IAM role that Vault will assume in order to validate authentication requests across AWS accounts. This option is only required in a cross-account authentication scenario. Please check [this documentation](https://developer.hashicorp.com/vault/docs/auth/aws#cross-account-access) out for more details.

**Comment by jeffery** [comment]
Thanks @petersin0422 . Yes it is for cross account authentication scenario. I am having issues getting it to work. Will read up on the documentation and see how I go.

**Reply by petersin0422** [reply]
Can you post your scenario here? Maybe I can help.

**Reply by jeffery** [reply]
The problem is when you use MFA + Cross account authentication. This solution works for me https://gist.github.com/Westixy/bc70ee782fe759094bf5c1c65c248f6c
